File: src/main/java/com/robohorse/robopojogenerator/generator/consts/annotations/KotlinAnnotations.java
package com.robohorse.robopojogenerator.generator.consts.annotations;

/**
 * This is the KotlinAnnotations class
 * Please put some info here.
 *
 * @author Wafer Li
 * @since 16/10/24 21:07
 */
public interface KotlinAnnotations {
    String INFO_ANNOTATION = "";

    interface GSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@field:SerializedName(\"%1$s\")";
    }

    interface LOGAN_SQUARE {
        String CLASS_ANNOTATION = INFO_ANNOTATION + "\n@JsonObject";
        String ANNOTATION = "@field:SerializedName(\"%1$s\")\n\t@field:JsonField(name = arrayOf(\"%1$s\"))";
    }

    interface JACKSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@field:JsonProperty(\"%1$s\")";
    }

    interface FAST_JSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@JSONField(name=\"%1$s\")";
    }

    interface MOSHI {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@Json(name=\"%1$s\")";
    }
}

File: generator/src/main/kotlin/com/robohorse/robopojogenerator/generator/consts/annotations/KotlinAnnotations.kt
package com.robohorse.robopojogenerator.generator.consts.annotations

internal sealed class KotlinAnnotations(
    val classAnnotation: String = EMPTY_ANNOTATION,
    val annotation: String
) {
    object GSON : KotlinAnnotations(
        annotation = "@field:SerializedName(\"%1\$s\")"
    )

    object LOGAN_SQUARE : KotlinAnnotations(
        classAnnotation = "@JsonObject",
        annotation = "@field:JsonField(name = [\"%1\$s\"])"
    )

    object JACKSON : KotlinAnnotations(
        annotation = "@field:JsonProperty(\"%1\$s\")"
    )

    object FAST_JSON : KotlinAnnotations(
        annotation = "@JSONField(name=\"%1\$s\")"
    )

    object MOSHI : KotlinAnnotations(
        annotation = "@Json(name=\"%1\$s\")"
    )
}

private const val EMPTY_ANNOTATION = ""
